src/GslibIO.jl

```diff
@@ -45,35 +49,98 @@ function load(file::File{format"GSLIB"})
 end
 
 """
-    load_legacy(filename, dims; origin=(0.,0.,0.), spacing=(1.,1.,1.), na=-999)
+    parse_gslib(filename::AbstractString)
 
-Load legacy GSLIB `filename` into a grid with `dims`, `origin` and `spacing`.
-Optionally set the value used for missings `na`.
+Helper function to parse a standard GSLIB file.
+It returns the data as a matrix `X`, `header`, number of variables `nvars`,
+extra info in the second line `extrainfo`, and the variable names as String `vars`.
 """
-function load_legacy(filename::AbstractString, dims::NTuple{3,Int};
-                     origin=(0.,0.,0.), spacing=(1.,1.,1.), na=-999)
+function parse_gslib(filename::AbstractString)
   open(filename) do fs
     # skip header
-    readline(fs)
+    header = readline(fs)
 
-    # number of properties
-    nvars = parse.(Int, readline(fs))
+    # number of properties is the first integer (it may contain extra info)
+    line = readline(fs)
+    linecont = split(line)
+    nvars = parse(Int, linecont[1])
+    extrainfo = linecont[1:end]
 
     # properties names
-    vars = [Symbol(readline(fs)) for i in 1:nvars]
+    vars = [strip(readline(fs)) for i in 1:nvars]
 
     # read property values
     X = readdlm(fs)
 
-    # handle missing values
-    replace!(X, na=>NaN)
```

Comment:
Simply because the parser should not take such action. The parse just parses the content as the file is.
You should be able to see the missing values handled in lines 94 and 134 respectively, which can be improved to avoid duplication of code. 

Remember that the previous code only could parse grid (and with a bug). To make the load/save generic is worthy to have the parser as a separated function.

Reply:
I like this. Fully agree that the parser should just consume the raw data and that we need to post-process it later to convert missing values for example. I am looking forward to seeing this clean `parse_gslib` function before we can continue reviewing the other modifications proposed here. Remember that this function should return a well-documented struct with the GSLIB spec. I still think that the spec is different for point set and grid, so I am looking forward to learn from you this single spec.